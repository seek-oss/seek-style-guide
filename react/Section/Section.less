@import (reference) "~seek-style-guide/theme";

@desktop-gutter: (@gutter-width * 1.25);

.root {
  padding: (@row-height * 3) @gutter-width;
  @media @desktop {
    padding-left: @desktop-gutter;
    padding-right: @desktop-gutter;
  }

  &.pullout {
    margin: 0 (-@gutter-width);
    @media @desktop {
      margin: 0 -@desktop-gutter;
    }
  }

  &.slim {
    padding-top: 0;
    padding-bottom: 0;
  }

  &.positive {
    color: @sk-positive;

    &.primary {
      background-color: @sk-positive;
      color: @sk-white;
    }

    &.secondary {
      background-color: @sk-positive-light;
    }
  }

  &.info {
    color: @sk-info;

    &.primary {
      background-color: @sk-info;
      color: @sk-white;
    }

    &.secondary {
      background-color: @sk-info-light;
    }
  }

  &.critical {
    color: @sk-critical;

    &.primary,
    &.secondary {
      background-color: @sk-critical-light;
      color: @sk-critical;
    }
  }

  &.help {
    color: @sk-help;

    &.primary,
    &.secondary {
      background-color: @sk-help-light;
      color: @sk-help;
    }
  }
}

.header {
  padding-top: (@row-height * 5);
  padding-bottom: (@row-height * 5);
  @media @desktop {
    padding-top: (@row-height * 7);
    padding-bottom: (@row-height * 6);
  }
}
