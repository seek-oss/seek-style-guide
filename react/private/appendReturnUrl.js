export default (url, returnUrl) => {
  const urlSuffix = returnUrl
    ? `?returnUrl=${encodeURIComponent(returnUrl)}`
    : '';

  return `${url}${urlSuffix}`;
};
