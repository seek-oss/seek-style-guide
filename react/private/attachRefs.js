// @flow
export default (...refs: Array<Function>) => (
  ref: ?React$ElementRef<'input'> | ?React$ElementRef<'textarea'>
): void => {
  refs.forEach(callRef => {
    if (typeof callRef === 'function') {
      callRef(ref);
    }
  });
};
