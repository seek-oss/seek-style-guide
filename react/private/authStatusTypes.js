export const AUTHENTICATED = 'authenticated';
export const UNAUTHENTICATED = 'unauthenticated';
export const AUTH_PENDING = 'pending';
