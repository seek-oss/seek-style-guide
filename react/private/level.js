// @flow
export type Level = 'primary' | 'secondary' | 'tertiary';

export const LEVEL: {
  ['PRIMARY' | 'SECONDARY' | 'TERTIARY']: Level
} = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  TERTIARY: 'tertiary'
};
