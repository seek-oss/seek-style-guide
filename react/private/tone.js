// @flow
type Tone = 'positive' | 'info' | 'critical' | 'help' | 'neutral';

export const TONE: {
  ['POSITIVE' | 'INFO' | 'CRITICAL' | 'HELP' | 'NEUTRAL']: Tone
} = {
  POSITIVE: 'positive',
  INFO: 'info',
  CRITICAL: 'critical',
  HELP: 'help',
  NEUTRAL: 'neutral'
};
