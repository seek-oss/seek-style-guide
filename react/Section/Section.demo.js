import React from 'react';
import PropTypes from 'prop-types';
import Section from './Section';
import PageBlock from '../PageBlock/PageBlock';
import Text from '../Text/Text';
import { TONE } from '../private/tone';
import { LEVEL } from '../private/level';

const SectionContainer = ({ component: DemoComponent, componentProps }) => (
  <PageBlock>
    <div style={{ backgroundColor: 'aqua' }}>
      <DemoComponent {...componentProps} />
    </div>
  </PageBlock>
);

const SectionUnstyledContent = (
  <Text>This content is nested within a Section component</Text>
);

SectionContainer.propTypes = {
  component: PropTypes.any,
  componentProps: PropTypes.object.isRequired
};

export default {
  route: '/section',
  title: 'Section',
  category: 'Layout',
  component: Section,
  container: SectionContainer,
  block: true,
  initialProps: {
    slim: false,
    pullout: false,
    onClose: () => console.log('On close handler called'),
    children: (
      <div style={{ backgroundColor: 'blue', color: 'white' }}>
        <Text>This content is nested within a Section component</Text>
      </div>
    )
  },
  options: [
    {
      label: 'States',
      type: 'checklist',
      states: [
        {
          label: 'Header',
          transformProps: props => ({
            ...props,
            header: true,
            children: (
              <div style={{ backgroundColor: 'blue', color: 'white' }}>
                <Text hero>
                  This content is nested within a Section component
                </Text>
              </div>
            )
          })
        },
        {
          label: 'Slim',
          transformProps: props => ({
            ...props,
            slim: true
          })
        },
        {
          label: 'Pullout',
          transformProps: props => ({
            ...props,
            pullout: true
          })
        }
      ]
    },
    {
      label: 'Tone',
      type: 'radio',
      states: [
        {
          label: 'Select tone',
          transformProps: props => props
        },
        {
          label: 'Positive',
          transformProps: props => ({
            ...props,
            tone: TONE.POSITIVE,
            children: SectionUnstyledContent
          })
        },
        {
          label: 'Info',
          transformProps: props => ({
            ...props,
            tone: TONE.INFO,
            children: SectionUnstyledContent
          })
        },
        {
          label: 'Critical',
          transformProps: props => ({
            ...props,
            tone: TONE.CRITICAL,
            children: SectionUnstyledContent
          })
        },
        {
          label: 'Help',
          transformProps: props => ({
            ...props,
            tone: TONE.HELP,
            children: SectionUnstyledContent
          })
        }
      ]
    },
    {
      label: 'Level',
      type: 'radio',
      states: [
        {
          label: 'Select level',
          transformProps: props => props
        },
        {
          label: 'Primary',
          transformProps: props => ({
            ...props,
            level: LEVEL.PRIMARY,
            children: SectionUnstyledContent
          })
        },
        {
          label: 'Secondary',
          transformProps: props => ({
            ...props,
            level: LEVEL.SECONDARY,
            children: SectionUnstyledContent
          })
        }
      ]
    }
  ]
};
