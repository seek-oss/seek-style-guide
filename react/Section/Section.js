// @flow
import classnames from 'classnames';
import React from 'react';

import styles from './Section.less';
import { TONE } from '../private/tone';
import type { Level } from '../private/level';

type Props = {
  children: any,
  className: string,
  header: boolean,
  pullout: boolean,
  slim: boolean,
  tone:
    | typeof TONE.POSITIVE
    | typeof TONE.CRITICAL
    | typeof TONE.INFO
    | typeof TONE.HELP
    | '',
  level: Level | ''
};

export default function Section({
  children,
  className,
  header,
  pullout,
  slim,
  tone,
  level,
  ...restProps
}: Props) {
  return (
    <div
      {...restProps}
      className={classnames({
        [className]: className,
        [styles.root]: true,
        [styles.header]: header,
        [styles.pullout]: pullout,
        [styles.slim]: slim,
        [styles[tone]]: tone,
        [styles[level]]: level
      })}
    >
      {children}
    </div>
  );
}

Section.defaultProps = {
  className: '',
  header: false,
  pullout: false,
  slim: false,
  tone: '',
  level: ''
};
