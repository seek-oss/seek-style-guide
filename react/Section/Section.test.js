import React from 'react';
import { shallow } from 'enzyme';
import Section from './Section';
import { TONE } from '../private/tone';
import { LEVEL } from '../private/level';

const renderSection = props =>
  shallow(<Section {...props}>Test content</Section>);

describe('Section:', () => {
  describe('types:', () => {
    it('should render positive sections', () => {
      const section = renderSection({ tone: TONE.POSITIVE });
      expect(section).toMatchSnapshot();
    });

    it('should render info sections', () => {
      const section = renderSection({ tone: TONE.INFO });
      expect(section).toMatchSnapshot();
    });

    it('should render critical sections', () => {
      const section = renderSection({ tone: TONE.CRITICAL });
      expect(section).toMatchSnapshot();
    });

    it('should render help sections', () => {
      const section = renderSection({ tone: TONE.HELP });
      expect(section).toMatchSnapshot();
    });
  });

  describe('levels:', () => {
    it('should render primary sections', () => {
      const section = renderSection({ level: LEVEL.PRIMARY });
      expect(section).toMatchSnapshot();
    });

    it('should render secondary sections', () => {
      const section = renderSection({ level: LEVEL.SECONDARY });
      expect(section).toMatchSnapshot();
    });
  });

  it('should render the pullout style', () => {
    const section = renderSection({ pullout: true });
    expect(section).toMatchSnapshot();
  });

  it('should render the slim style', () => {
    const section = renderSection({ slim: true });
    expect(section).toMatchSnapshot();
  });

  it('should render the header style', () => {
    const section = renderSection({ header: true });
    expect(section).toMatchSnapshot();
  });

  it('should accept additional classnames and place them on the root', () => {
    const section = renderSection({
      level: LEVEL.SECONDARY,
      className: 'testClass'
    });
    expect(section).toMatchSnapshot();
  });

  it('should accept additional props', () => {
    const section = renderSection({ testProp: true });
    expect(section).toMatchSnapshot();
  });

  it('should render multiple classes together', () => {
    const section = renderSection({
      level: LEVEL.SECONDARY,
      tone: TONE.INFO,
      slim: true
    });
    expect(section).toMatchSnapshot();
  });
});
