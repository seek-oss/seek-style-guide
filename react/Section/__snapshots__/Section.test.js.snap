// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section: levels: should render primary sections 1`] = `
<div
  className="Section__root Section__primary"
>
  Test content
</div>
`;

exports[`Section: levels: should render secondary sections 1`] = `
<div
  className="Section__root Section__secondary"
>
  Test content
</div>
`;

exports[`Section: should accept additional classnames and place them on the root 1`] = `
<div
  className="testClass Section__root Section__secondary"
>
  Test content
</div>
`;

exports[`Section: should accept additional props 1`] = `
<div
  className="Section__root"
  testProp={true}
>
  Test content
</div>
`;

exports[`Section: should render multiple classes together 1`] = `
<div
  className="Section__root Section__slim Section__info Section__secondary"
>
  Test content
</div>
`;

exports[`Section: should render the header style 1`] = `
<div
  className="Section__root Section__header"
>
  Test content
</div>
`;

exports[`Section: should render the pullout style 1`] = `
<div
  className="Section__root Section__pullout"
>
  Test content
</div>
`;

exports[`Section: should render the slim style 1`] = `
<div
  className="Section__root Section__slim"
>
  Test content
</div>
`;

exports[`Section: types: should render critical sections 1`] = `
<div
  className="Section__root Section__critical"
>
  Test content
</div>
`;

exports[`Section: types: should render help sections 1`] = `
<div
  className="Section__root Section__help"
>
  Test content
</div>
`;

exports[`Section: types: should render info sections 1`] = `
<div
  className="Section__root Section__info"
>
  Test content
</div>
`;

exports[`Section: types: should render positive sections 1`] = `
<div
  className="Section__root Section__positive"
>
  Test content
</div>
`;
